### PR TITLE
Vectorizer naming update

### PR DESCRIPTION
**Pull request by billy-the-fish**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -15,19 +15,22 @@ pgai simplifies the process of building [search](https://en.wikipedia.org/wiki/S
...
-* Reason over your data and facilitate use cases like [classification, summarization, and data enrichment](docs/advanced.md) on your existing relational data in PostgreSQL.
+* [Create vector embeddings for your data](/docs/vectorizer-quick-start.md).
+* Retrieve LLM chat completions from models like [Claude Sonnet 3.5](/docs/anthropic.md), [OpenAI GPT4o](/docs/openai.md), [Cohere Command](/docs/cohere.md), and [Llama 3 (via Ollama)](/docs/ollama.md).
+* Reason over your data and facilitate use cases like [classification, summarization, and data enrichment](/docs/openai.md) on your existing relational data in PostgreSQL.
 
 Here's how to get started with pgai:
 
+* **TL;DR**:
+  * [Test drive automatic embedding vectorization](/docs/vectorizer-quick-start.md): quicky create embeddings using 
+     a pre-built Docker developer environment with a self-hosted Postgres instance with pgai, vectorizer, and vectorizer worker installed.
```
Comment (Askir) [inline README.md:26]:
```suggestion
* **TL;DR**:
  * [Try out automatic embedding vectorization](/docs/vectorizer-quick-start.md): quicky create embeddings using 
     a pre-built Docker developer environment with a self-hosted Postgres instance with pgai and our vectorizer worker installed. (<10 minutes if you have docker installed!)
```

Not quite sure about the use of the term "test drive" here?
Test driven development does not seem relevant to the quick start, and is a loaded term so quite confusing imo.

File: docs/vectorizer-quick-start.md [hunk 2]
```diff
@@ -0,0 +1,130 @@
+# Vectorizer quick start
+
+This page shows you how to create a vectorizer in a self-hosted Postgres instance, then use 
+a pgai vectorizer worker to create embeddings from data in your database. You view the 
+embeddings with a standard SQL query.
```
Comment (Askir) [inline docs/vectorizer-quick-start.md:5]:
```suggestion
This page shows you how to create a vectorizer in a self-hosted Postgres instance, then use 
the pgai vectorizer worker to create embeddings from data in your database. To finish off we show how simple it is to do a semantic search query on the embedded data in one query!
```

Viewing the embeddings is kinda lame, we actually do semantic search which is imo the aha effect of this guide.

File: docs/vectorizer-quick-start.md [hunk 3]
```diff
@@ -0,0 +1,130 @@
+# Vectorizer quick start
+
+This page shows you how to create a vectorizer in a self-hosted Postgres instance, then use 
+a pgai vectorizer worker to create embeddings from data in your database. You view the 
+embeddings with a standard SQL query. 
+
+## Prerequisites
+
+To follow the steps in this page, you need to:
+- [Create a local developer environment](/docs/vectorizer-worker.md#install-the-timescaledb-ha-docker-image)
```
Comment (Askir) [inline docs/vectorizer-quick-start.md:10]: Not a big fan of making people jump around different pages so much. I liked it more when the compose was in here. I also think the compose should not be used for anything else, its way too minimal for production use so having it in a more general file is just confusing.

File: docs/vectorizer-quick-start.md [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+   - Docker: `docker exec -it pgai-db-1 psql -U postgres`
+   - psql:  `psql postgres://postgres:postgres@localhost:5432/postgres`
+
+1. **Enable pgai on your database**
+
+    ```sql
+    CREATE EXTENSION IF NOT EXISTS ai CASCADE;
+    ```
+
+1. **Create the `blog` table with the following schema**
```
Comment (Askir) [inline docs/vectorizer-quick-start.md:25]:
```suggestion
1. **Create a `blog` table with the following schema**
```

File: docs/vectorizer-quick-start.md [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+     ✔ 38e39325ccaf Pull complete                                                                                                                                        6.5s 
+     ✔ 8d50dded1d76 Pull complete                                                                                                                                        6.5s 
+     ✔ 5a73fecc1c3f Pull complete                                                                                                                                        6.5s 
+     ✔ 03f1e7e22d43 Pull complete                                                                                                                                       20.0s 
+     ✔ 33c61f413670 Pull complete                                                                                                                                       20.0s 
+     ✔ dc491244f87c Pull complete                                                                                                                                       20.0s 
+     [+] Running 1/1
+     [+] Running 1/2ai-vectorizer-worker-1  
+    ```
```
Comment (Askir) [inline docs/vectorizer-quick-start.md:88]: Is this shell output relevant? We don't have it for starting the db container, I think it just bloats the tutorial a bit.
As a user I might skim through the steps of such a quick start and if it looks to complicated I just wont do it. I think having a lot of console output contributes to that.

File: docs/vectorizer-quick-start.md [hunk 6]
```diff
@@ -0,0 +1,130 @@
...
+   ```
+
+   You see the vectorizer worker pick up the table and process it.
+   ```shell
+    vectorizer-worker-1  | 2024-10-23 12:56:36 [info     ] running vectorizer             vectorizer_id=1
+    ```
+   
+    To see the status of your vectorizer queue, use the following query 
+    `select ai.vectorizer_queue_pending(1) \watch`. To see the the database logs run 
+    `docker-compose logs -f db`.
```
Comment (Askir) [inline docs/vectorizer-quick-start.md:102]: Imo not relevant, same argument as previous comment. It just delays the aha effect of the next step.

File: docs/vectorizer-worker.md [hunk 7]
```diff
@@ -0,0 +1,234 @@
+
+# Run vectorizers in a self-hosted database using pgai vectorizer worker
```
Comment (Askir) [inline docs/vectorizer-worker.md:2]: This file is a bit confusing now, it previously only covered the python vectorizer worker and now also contains database hosting? Is it supposed to be just a general self hosting guide? Then I think we should expand and structure it in two parts: 
1. The database
2. The vectorizer worker

File: docs/vectorizer.md [hunk 8]
```diff
@@ -26,7 +26,8 @@ Our system empowers you to:
 
 This page offers a comprehensive overview of Vectorizer features,
 demonstrating how it streamlines the process of working with vector embeddings
-in your database. For a more detailed technical specification, see the
+in your database. To quickly test drive embeddings using a pre-built Docker developer environment, see the 
+[Vectorizer quick start](/docs/vectorizer-quick-start.md). For a more detailed technical specification, see the
```
Comment (Askir) [inline docs/vectorizer.md:30]:
```suggestion
in your database. To quickly try out embeddings using a pre-built Docker developer environment, see the 
[Vectorizer quick start](/docs/vectorizer-quick-start.md). For a more detailed technical specification, see the
```

File: docs/vectorizer.md [hunk 9]
```diff
@@ -62,11 +63,11 @@ Timescale Cloud or on a self-hosted Postgres server.
 
 - Self-hosted Postgres
 
-  Set an environment variable that is the [same as your API key name](./self-hosting-vectorizer.md#install-and-configure-vectorizer-cli). 
+  Set an environment variable that is the [same as your API key name](./vectorizer-worker.md#install-and-configure-vectorizer-worker). 
   For example:
   ```bash
   export OPENAI_API_KEY="Your OpenAI API key"
-  vectorizer
+  pgai vectrorizer worker
```
Comment (Askir) [inline docs/vectorizer.md:70]:
```suggestion
```
What is this call of vectorizer for here? The `vectorizer.md` explains the SQL vectorizer interface. It has nothing to do with the worker and calling it here is not necessary and just confusing to me.

File: docs/vectorizer.md [hunk 10]
```diff
@@ -144,9 +149,12 @@ SELECT
    chunk,
    embedding <=> <query embedding> as distance
```
Comment (Askir) [inline docs/vectorizer.md:150]:
```suggestion
   embedding <=> ai.openai_embed(...) as distance
```

Since we are now adding secret fetching to cloud we can do the openai embedding call here. This is much nicer because it allows copy pasting the docs and actually doing semantic search.

Can be a separate PR though to change this.

**Review by Askir — COMMENTED** [review]
Leaving a very opinionated review here :D

Overall I like the changes though!

File: docs/vectorizer-quick-start.md (hunk 5 above)
Comment (billy-the-fish) [reply docs/vectorizer-quick-start.md]: Removed

File: docs/vectorizer-quick-start.md (hunk 6 above)
Comment (billy-the-fish) [reply docs/vectorizer-quick-start.md]: Removed

File: docs/vectorizer-quick-start.md (hunk 3 above)
Comment (billy-the-fish) [reply docs/vectorizer-quick-start.md:9]: I am in 2 minds. I agree that having all in one place for a quickstart is easier. However, you get different instructions and then there is no consistent way of working. For example, the docker image is the standard Timescale dev environment, and the install guide is here: https://docs.timescale.com/self-hosted/latest/install/installation-docker/. 

In terms of this repo I would prefer a single source of truth and have all install guides here: https://github.com/timescale/pgai/blob/vectorizer-naming-update/README.md#installation.

File: docs/vectorizer-worker.md [hunk 11]
```diff
@@ -0,0 +1,234 @@
...
+
+  
+### Install the vectorizer worker Docker image
+
+This docker image supplies a pgai image with vectorizer worker. You use this image to 
+run vectorizers on any self-hosted Postgres database that has the pgai extension activated.
+
+On your local machine:
+
+IAIN: Alejandro, I'm not too sure what to write here.
```
Comment (alejandrodnm) [inline docs/vectorizer-worker.md:89]: Since this part ties to the quick start, it should be the rest of the docker-compose file, the part that defines the `vectorizer-worker` service. Otherwise the quickstart guide won't work.

You should copy:

https://github.com/timescale/pgai/pull/153/files#diff-b8b30779c88ab44c3d6a4e3dd2d16f7149f47052527d5e428b4730a4c5ea1244R25-R29

But change the `VECTORIZER_DB_URL`  to `PGAI_VECTORIZER_WORKER_DB_URL`

Then the other question is, should we also give a plain Docker command like:

```
docker run timescale/pgai-vectorizer-worker:0.1.0rc4 --db-url $DB_URL
```

File: docs/vectorizer-quick-start.md (hunk 3 above)
Comment (billy-the-fish) [reply docs/vectorizer-quick-start.md]: I've put the dev env back in the quickstart. The words didn't quite work in the run page.

File: docs/vectorizer-worker.md [hunk 12]
```diff
@@ -0,0 +1,234 @@
...
+1. **Run the vectorizer worker**
+
+    After you [define a vectorizer in your database](/docs/vectorizer.md#define-a-vectorizer), you run 
+    a vectorizer worker to generate and update your embeddings:
+
+    1. Set the connection string to your self-hosted database and the API key for the external embeddings
+      provider as environment variables:
+
+       ```bash
+       export VECTORIZER_DB_URL="postgres://<user>:<password>@<host>:<port>/<dbname>"
```
[the remark this answers is not in the document]
Comment (billy-the-fish) [reply docs/vectorizer-worker.md]: Done

File: docs/vectorizer-worker.md [hunk 13]
```diff
@@ -0,0 +1,234 @@
...
+  * Local environment: [Python3][python3] and [pip][pip]
+  * All environments: A Postgres client like [psql][psql]
+* Create a key for you AI provider:
+  * [OpenAI][openai-key]
+
+## Install and configure vectorizer worker
+
+To be able to run vectorizers in your self-hosted database, either:
+
+- [Install the timescaledb-ha Docker image](#install-the-timescaledb-ha-docker-image): a Docker image containing a complete local developer environment
```
[the remark this answers is not in the document]
Comment (billy-the-fish) [reply docs/vectorizer-worker.md]: Updated.

File: docs/vectorizer-worker.md (hunk 11 above)
Comment (billy-the-fish) [reply docs/vectorizer-worker.md]: Done. 
Added the run command.

File: docs/vectorizer-quick-start.md [hunk 14]
```diff
@@ -0,0 +1,108 @@
...
+
+This page shows you how to create a vectorizer in a self-hosted Postgres instance, then use 
+the pgai vectorizer worker to create embeddings from data in your database. To finish off we show how simple it is to do a semantic search query on the embedded data in one query!
+
+## Prerequisites
+
+To follow the steps in this page, you need to:
+- [Create a local developer environment](/docs/vectorizer-worker.md#install-the-timescaledb-ha-docker-image)
+
+## Create, run and query a vectorizer
```
[the remark this answers is not in the document]
Comment (billy-the-fish) [reply docs/vectorizer-quick-start.md]: Updated. Thanks.

File: docs/vectorizer-worker.md [hunk 15]
```diff
@@ -0,0 +1,264 @@
...
+worker instance. For `docker compose` you add arguments using either the `command` or `environment`
+flags in `docker-compose.yml`. 
+
+A vectorizer worker can:
+
+- Run a single vectorizer:
+
+  To run the vectorizer with id 42:
+  - local: `pgai vectorizer worker -i 42`
+  - Docker: `command: ["-i", "42"]`
```
Comment (alejandrodnm) [inline docs/vectorizer-worker.md:183]: This section for `Docker` it's actually docker compose.

I'd change that to docker-compose. 

Docker could be something like `docker run timescale/pgai-vectorizer-worker:{tag version} -i 42`

Comment (billy-the-fish) [reply docs/vectorizer-worker.md]: Added the docker run commands.
